cobra/flux_analysis/loopless.py

```diff
@@ -1,6 +1,232 @@
 from ..core import Reaction, Metabolite
 from ..manipulation.modify import convert_to_irreversible
 from six import iteritems
+from cobra.util import add_to_solver
+from sympy.core.singleton import S
+
+try:
+    import numpy
+    from cobra.util import nullspace
+except ImportError:
+    numpy = None
+    nullspace = None
+
+
+def add_loopless(model, zero_cutoff=1e-12):
+    """Add variables and constraints to make a model thermodynamically
+    feasible. This removes flux loops. The used formulation is described
+    in [1]_.
+
```

Comment:
perhaps could give an example of when to use `add_loopless` versus `loopless_solution` to help understand the reason for both of them?